### PR TITLE
Fix Niconico Series extractor

### PR DESCRIPTION
**Pull request by sqrtNOT**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)

---

Essentially, the niconico series page went from using relative references to using absolute references so the current regex wasn't finding any video urls on the test cases.

### CODE REVIEW

**Comment by gamer191** [comment]
NOTE: please ignore the failing commit test. It's due to an external bug (the external bug being that pytest is partially broken in python 3.11-dev)

File: yt_dlp/extractor/niconico.py [hunk 1]
```diff
@@ -647,14 +647,14 @@ class NiconicoSeriesIE(InfoExtractor):
             'id': '110226',
             'title': 'ご立派ァ！のシリーズ',
         },
-        'playlist_mincount': 10,  # as of 2021/03/17
+        'playlist_mincount': 10,  # as of 2022/06/01
```
Comment (pukkandan) [inline yt_dlp/extractor/niconico.py:650]:
```suggestion
        'playlist_mincount': 10,
```

File: yt_dlp/extractor/niconico.py [hunk 2]
```diff
@@ -647,14 +647,14 @@ class NiconicoSeriesIE(InfoExtractor):
...
     }, {
         'url': 'https://www.nicovideo.jp/series/12312/',
         'info_dict': {
             'id': '12312',
             'title': 'バトルスピリッツ　お勧めカード紹介(調整中)',
         },
-        'playlist_mincount': 97,  # as of 2021/03/17
+        'playlist_mincount': 103,  # as of 2022/06/01
```
Comment (pukkandan) [inline yt_dlp/extractor/niconico.py:657]:
```suggestion
        'playlist_mincount': 103,
```

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/niconico.py [hunk 3]
```diff
@@ -672,7 +672,7 @@ def _real_extract(self, url):
             title = unescapeHTML(title)
         playlist = [
             self.url_result(f'https://www.nicovideo.jp/watch/{v_id}', video_id=v_id)
-            for v_id in re.findall(r'href="/watch/([a-z0-9]+)" data-href="/watch/\1', webpage)]
+            for v_id in re.findall(r'data-href=[\'"](?:https://www\.nicovideo\.jp)?/watch/([a-z0-9]+)', webpage)]
```
Comment (pukkandan) [inline yt_dlp/extractor/niconico.py:675]: Shouldn't we also look for the old pattern? Or are you sure all pages have transitioned?

**Pull request merged**